README.md

```diff
@@ -115,42 +113,93 @@ mvn install #inside the spring-search folder
+import org.springframework.data.rest.core.annotation.RepositoryRestResource;
+
+@RepositoryRestResource
+public interface yourRepository extends JpaRepository<yourModel, Integer>, JpaSpecificationExecutor<yourModel> {
+
+}
+```
+
+Import the library in your controller
 ```kotlin
 import com.sipios.springsearch.anotation.SearchSpec;
 ```
 
-6. Use it (see [Usage](#usage) for various examples)
+Use it in your controller
 ```kotlin
-@GetMapping("someMapping")
+@GetMapping("searchUrl")
 fun yourFunctionNameHere(@SearchSpec specs: Specification<yourModelHere>): ResponseEntity<yourResponse> {
     return ResponseEntity(yourRepository.findAll(Specification.where(specs)), HttpStatus.OK)
 }
 ```
 
+1. Using the equal operator `:`  
+Request : `/cars?search=color:Red`  
+![equal operator example](./docs/images/equal-example.gif)
 
-<!-- USAGE EXAMPLES -->
-## Usage
+2. Using the not equal operator `!`  
+Request : `/cars?search=color!Red`
+![not equal operator example](./docs/images/not-equal-example.gif)
 
-Use this space to show useful examples of how a project can be used. Additional screenshots, code examples and demos work well in this space. You may also link to more resources.
+3. Using the greater than operator `>`  
+Request : `/cars?search=creationyear>2017`  
+![greater than operator example](./docs/images/greater-than-example.gif)
 
-1. Using the equal operator `:`
-![equal operator example](./docs/images/equal-example.gif)
+4. Using the less than operator `<`  
+Request : `/cars?search=price<100000`  
+![less than operator example](./docs/images/less-than-example.gif)
 
-2. Using the not equal operator `!`
-![not equal operator example](./docs/images/not-equal-example.gif)
+5. Using the starts with operator `*`  
+*For the ends with operator, simply place `*` at the beginning of the word*.  
+*For the contains operator, simply place `*` at the beginning and the end of the word*.  
+Request : `/cars?search=brand:Aston*`  
+![starts with operator example](./docs/images/starts-with-example.gif)
+
+6. Using the `OR` operator  
+Request : `/cars?search=color:Red OR color:Blue`  
+![or operator example](./docs/images/or-example.gif)
+
+7. Using the `AND` operator  
+Request : `/cars?search=brand:Aston* AND price<300000`  
+![and operator example](./docs/images/and-example.gif)
+
+8. Using parenthesis  
+Request : `/cars?search=( brand:Nissan OR brand:Chevrolet ) AND color:Blue`  
+*Note: Space inside parenthesis is not necessary*
+![parenthesis example](./docs/images/parenthesis-example.gif)
+
+9. Using space in nouns  
+Request : `/cars?search=model:'Spacetourer Business Lounge'`  
+![space example](./docs/images/space-example.gif)
 
-_For more examples, please refer to the [Documentation](https://example.com)_
+10. Using special characters  
+Request: `/cars?search=model:中华V7`
+![special characters example](./docs/images/special-characters-example.gif)
 
+11. Using deep fields  
+Request : `/cars?search=options.transmission:Auto`
```

Comment:
You don't give a warning about deep fields?

Reply:
No, they work well on strings